features/admin.claims.v1/components/edit/local-claim/edit-additional-properties-local-claims.tsx

```diff
@@ -101,13 +141,27 @@ export const EditAdditionalPropertiesLocalClaims:
                                 ) }
                                 requiredField={ true }
                                 update={ (data: KeyValue[]) => {
+                                    const hasRestrictedKey: boolean = data.some(
+                                        (item: KeyValue) => item.key === RESTRICTED_PROPERTY_KEYS[0]
+                                    );
+
+                                    setShowWarning(hasRestrictedKey);
+
+                                    const filteredData: KeyValue[] = data.filter(
+                                        (item: KeyValue) => item.key !== RESTRICTED_PROPERTY_KEYS[0]
+                                    );
```

Comment:
Redundant filtering here. You're calling data.filter to create filteredData even if the restricted key exists. However, the function returns early if hasRestrictedKey is true, meaning filteredData is created unnecessarily.

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/7106/commits/a5fb2e912007474c3ca67b3f4466fb16eb19b772